kqueen/engines/jenkins.py

```diff
@@ -74,39 +106,40 @@ def _get_provision_job_builds(self):
         """
         return self.client.get_job_info(self.provision_job_name, depth=1)
 
-    @classmethod
-    def engine_status(cls, **kwargs):
+    def _parameter_exist(self, parameter):
         """
-        Implementation of :func:`~kqueen.engines.base.BaseEngine.engine_status`
-        """
-        conn_kw = {
-            'username': kwargs.get('username', config.get('JENKINS_USERNAME')),
-            'password': kwargs.get('password', config.get('JENKINS_PASSWORD')),
-            'timeout': 10
-        }
-        status = config.get('PROVISIONER_UNKNOWN_STATE')
-        try:
-            client = jenkins.Jenkins(config.get('JENKINS_API_URL'), **conn_kw)
-            auth_verify = client.get_whoami()
-            if auth_verify:
-                status = config.get('PROVISIONER_OK_STATE')
-        except Exception as e:
-            logger.exception('Could not contact JenkinsEngine backend: ')
-            status = config.get('PROVISIONER_ERROR_STATE')
-        return status
+        Check, that defined key parameter exist in Job
 
-    def _get_client(self):
-        """
-        Initialize Jenkins client
+        Args:
+            string: 'ParameterName'
 
         Returns:
-            :obj:`jenkins.Jenkins`: initialized Jenkins client
+            bool: True if all parameters exist
         """
-        return jenkins.Jenkins(self.jenkins_url, **{
-            'username': self.username,
-            'password': self.password,
-            'timeout': 10
-        })
+
+        job_body = self.client.get_job_info(self.provision_job_name, depth=1)
```

Comment:
I suggest to add try-catch here and return False from function end at catch block